rsocket-transport-ktor/src/jvmTest/kotlin/io/rsocket/kotlin/TcpTransportTest.kt

```diff
@@ -18,20 +18,51 @@ package io.rsocket.kotlin
 
 import io.ktor.network.selector.*
 import io.ktor.network.sockets.*
+import io.ktor.util.*
 import io.rsocket.kotlin.connection.*
-import io.rsocket.kotlin.core.*
 import kotlinx.coroutines.*
+import kotlin.test.*
 
+@OptIn(InternalAPI::class)
 class TcpTransportTest : TransportTest() {
-    override suspend fun init(): RSocket = builder.connect("127.0.0.1", 2323).connection.connectClient()
+    private val selector = SelectorManager(Dispatchers.IO)
+    private val builder = aSocket(selector).tcp()
+    private lateinit var server: ServerSocket
 
-    companion object {
-        private val builder = aSocket(ActorSelectorManager(Dispatchers.IO)).tcp()
+    @BeforeTest
+    fun setup() {
+        server = runBlocking {
+            trySeveralTimes {
+                builder.bind("127.0.0.1", 2323)
```

Comment:
Fixed ports are always a problem.  Is it possible to use an ephemeral port (0) and read the actual port?

Reply:
Will check, if ktor support it

Reply:
We can always generate a random number on the range on our own

Reply:
Please no.  I think if this isn't possible it's a missing feature somewhere. This is pretty common pattern with port 0 then read actual port.